packages/eslint-plugin/src/rules/no-unnecessary-type-assertion.ts

```diff
@@ -128,6 +128,36 @@ export default createRule<Options, MessageIds>({
       );
     }
 
+    function isTypeUnchanged(uncast: ts.Type, cast: ts.Type): boolean {
+      if (uncast === cast) {
+        return true;
+      } else if (
```

Comment:
[Style] Nit: any reason to keep the indentation?

```suggestion
      }

```

https://github.com/JoshuaKGoldberg/create-typescript-app/issues/1267 tracks my interest in a rule for this. 😄